kv/dist_sender.go

```diff
@@ -527,6 +511,10 @@ func (ds *DistSender) Send(_ context.Context, call client.Call) {
 			reply.Header().Reset()
 			descNext = nil
 			desc, err := ds.rangeCache.LookupRangeDescriptor(args.Header().Key, options)
+			var leader proto.Replica
```

Comment:
because this is close to `desc`, which can be nil and that is really easy to forget.